foca/api/register_openapi.py

```diff
@@ -39,7 +37,13 @@ def register_openapi(
     # Iterate over OpenAPI specs
     for spec in specs:
         spec_modified = False
-        spec_parsed = ConfigParser.parse_yaml(spec.path)
+
+        # Parse & merge specs, if applicable
+        if isinstance(spec.path, str):
+            spec.path = [spec.path]
```

Comment:
@uniqueg  https://github.com/elixir-cloud-aai/foca/pull/75/files#diff-eb7d57a240e2ff03256e64aa7fb30a13L78  expects a string when specs are not modified. I have tested this. This will break when the input is like this:

```
api:
    specs:
        - path: 
            - petstore.yaml
          append: null
          connexion:
              strict_validation: True
              validate_responses: False
              options:
                  swagger_ui: True
                  serve_spec: True
```
I'll make the relevant changes along with unit tests.

Reply:
I don't understand this comment. What I implemented (I hope) should behave like this:
- when a single path (e.g., `- path: petstore.yaml`) is passed (`str`), it is converted to a list (`["petstore.yaml"]`) which is then passed to `.merge_yaml()`
- when a list of paths is passed, regardless of how many items are listed (like in your example; `List[str]`), it is passed to `.merge_yaml()` directly

Where exactly is the problem? I can see that there's no check what happens when an empty list is passed (this should be implemented), but I thought the single string is covered...

Reply:
If you see the line `78` here. You can see this code
`spec_use = spec.path`

`spec_use` expects a string when `spec_modified` is false. So when it is used after in `add.add_api`, it will break.

I covered this case in https://github.com/elixir-cloud-aai/foca/pull/75/files/1d01ebbd14f8cf63751b7c52141a86f0f4caa2bc#diff-eb7d57a240e2ff03256e64aa7fb30a13R45 though it was partially correct.

Reply:
Ah okay, got it. Well, easy to fix :)